### PR TITLE
Filters, authentication and corrections

### PR DESCRIPTION
**Pull request by sofiiaabdu**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -92,21 +106,32 @@ def random_picture_generator
     picture.save
   end
 
+  def filter(first = nil, city = nil, min = nil, max = nil, male = nil, female = nil)
+    return User.order(first_name: :"#{first}") if first
+    return User.where('age <= ? and age >= ?', max, min) if min && max
```
Comment (OlegLazaryev) [inline app/controllers/users_controller.rb]: Let's move this query into a scope

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -92,21 +106,32 @@ def random_picture_generator
...
+    return User.joins(:address).order("addresses.city #{city}") if city
```
Comment (OlegLazaryev) [inline app/controllers/users_controller.rb]: Let's move this query into a scope

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -92,21 +106,32 @@ def random_picture_generator
...
+    return User.male if male
+    return User.female if female
+    return User.all
```
Comment (OlegLazaryev) [inline app/controllers/users_controller.rb]: I would do a few improvements to this method. Let's discuss later today.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -92,21 +106,32 @@ def random_picture_generator
...
-    # Never trust parameters from the scary internet, only allow the white list through.
     def user_params
       params.require(:user).permit(:first_name, :last_name, :sex, :age, :about,
                                    address_attributes: [:zip, :city, :street, :house_member])
     end
+
+    def authorize_admin
+      return if current_user.try(:admin?)
+      redirect_to root_path, notice: 'Access denied!'
+    end
```
Comment (OlegLazaryev) [inline app/controllers/users_controller.rb]: Let's discuss this authorization method later today.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -70,43 +83,61 @@ def destroy
...
-    private
-    # Use callbacks to share common setup or constraints between actions.
     def set_user
-=begin
-      if params[:id] = 'sign_out'
-        sign_out current_user
-        redirect_to '/pictures'
-      else
-=end
+      if current_user.admin?
```
[the remark this answers is not in the document]
Comment (sofiiaabdu) [reply app/controllers/users_controller.rb]: Have explained in person.

**Comment by sofiiaabdu** [comment]
@OlegLazaryev Have fixed your remarks, please, check it again.

File: app/controllers/users_controller.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (OlegLazaryev) [reply app/controllers/users_controller.rb]: Thanks.

**Comment by OlegLazaryev** [comment]
@sofiiaabdu looking good. Thank you!
Additionally thank you for sticking to the initial specification of homework. It shows that you are reliable teammate. In production cases it is more important. Keep it.

**Pull request merged**